packages/api/src/controllers/managementPlanController.js

```diff
@@ -434,13 +436,29 @@ const managementPlanController = {
               [management_plan_id, taskIdsRelatedToManyManagementPlans],
             ));
 
-          return await ManagementPlanModel.query(trx)
+          const delPlan = await ManagementPlanModel.query(trx)
             .context(req.auth)
             .where({ management_plan_id })
             .delete();
+
+          if (delPlan) {
+            return deletedTasks;
+          } else {
+            return delPlan;
+          }
```

Comment:
@javiscript13 Thank you for taking over this task!
I don't think this will make any difference on the API itself. (What is returned here will be used on line 451 `if (result)`) What are these changes for??

Reply:
Hey @SayakaOno , thank you for reviewin my pull request!.

I'm not sure if I understand the question, but, on the quoted code, the `deletedTasks` array will be passed as a result to send the notifications if everything went ok (so `result` will have all the needed information to send notifications).  If there is an issue on the `ManagementPlanModel.query(trx)` that issue will be passed and status 404 will be sent by the api (as original).  

I did'nt find on knex documentation if a query fails does it sends a faulty value or a pure false, so I decided to pass that value.  

Do you think I should simplify the quoted code to return `delPlan?deletedTasks:false;` ?

Reply:
Is there a case `await ManagementPlanModel.query(trx)` succeeds, but `deletedTasks` becomes faulty? Is that what you were trying to catch?
When `delPlan` is truthy, I thought `result` will always be truthy too, so I asked the question! I might have missed something though...

Reply:
@SayakaOno,  the changes I'm proposing are doing the following:

I'm creating an array called `deletedTasks`, as there was a procedure to identify which tasks are related to the managentPlan that is going to be deleted, and as in this procedure all the information needed to fill the notifications is available (on the original PR this place was used to send the notifications), but as Duncan suggested if there was an error on the deletion the notifications would be already sent, .. so I'm just using this place to fill objects with the needed data to send notifications and pushing each object on the `deletedTasks` array, so the notifications are sent later when the transaction is confirmed to be ok.

Later in the code (line 440) is the query that deletes the management plan, and is the last procedure on the transaction.  So, if this query executes ok, I'm sending the `deletedTasks` as a result of the whole transaction if the deletion went ok it will send the notifications based on the gathered data (line 452) (on `deletedTasks`, and then in the other context as `result`).

The promise.all where the notifications are sent (line 453), has its own catch, so if there is an error sending the notifications it will only log the error, and will not affect the outer catch because that would inform the user that the managementPlan could not be deleted, but would be a mistake beacuse the managementePlan would be deleted but the error would be on the notification.

Reply:
Ahhhhhhhh! I'm so sorry, for some reason I kept missing `result.map`(line 454). I didn't notice `result` was being used there.
Let me buy you lunch 😭